misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java

```diff
@@ -320,6 +331,15 @@ private void execute() {
         });
     }
 
+    private boolean isTestDependsOnFailedFunctions(List<String> failedTests, List<String> dependantTests) {
+        for (String func : failedTests) {
+                if (dependantTests.contains(func)) {
+                    return true;
+                }
+        }
+        return false;
```

Comment:
this can be a one-liner;
return failedOrSkippedTests.stream().filter(dependentTests::contains).findAny().isPresent();

Reply:
Any special advantage by converting this into a single line? Actually we can make it shorter like below I guess. :)
````
failedOrSkippedTests.stream().anyMatch(dependentTests::contains); 
````

Reply:
+1, this is much better. It makes the code more readable and less number of lines is better.